tests/unit/src/test/scala/tests/ScalaVersionsSuite.scala

```diff
@@ -143,4 +143,59 @@ class ScalaVersionsSuite extends BaseSuite {
         V.scala3
     )
   }
+
+  test("compare-milestones-M2>M1") {
+    assert(
+      SemVer.isCompatibleVersion("3.0.0-M1", "3.0.0-M2")
+    )
+  }
+
+  test("compare-milestones-M1<M2") {
+    assert(
+      !SemVer.isCompatibleVersion("3.0.0-M2", "3.0.0-M1")
+    )
+  }
```

Comment:
These two tests will always be the same no? If one fails, the other should fail. If on passes the other should pass.

Reply:
I also switched the order of the parameters, so this are not strictly opposite tests. At least I have seen one fail and the other not 😅

Reply:
Ach! I know how it's called. I am actually testing transitivity!